Zend/zend_operators.h

```diff
@@ -482,6 +482,7 @@ ZEND_API void ZEND_FASTCALL zend_locale_sprintf_double(zval *op ZEND_FILE_LINE_D
 
 #define convert_scalar_to_number_ex(pzv)							\
 	if (Z_TYPE_P(pzv)!=IS_LONG && Z_TYPE_P(pzv)!=IS_DOUBLE) {		\
+		php_error_docref(NULL, E_DEPRECATED, "Method will produce a TypeError in PHP 8 when strict_types are enabled"); \
```

Comment:
Can you check if `strict_types` is enabled and only log if it is?

Reply:
I can't see an obvious way to do this without allocating memory for the configuration check which I'd like to avoid.